src/gcovr/formats/gcov/parser/json.py

```diff
@@ -87,26 +87,25 @@ def parse_coverage(
         if is_file_excluded(fname, include_filters, exclude_filters):
             continue
 
-        if file["file"] == "<stdin>":
-            message = f"Got sourcefile {file['file']}, using empty lines."
-            LOGGER.info(message)
-            source_lines = [b"" for _ in range(file["lines"][-1]["line_number"])]
-            source_lines[0] = f"/* {message} */".encode()
-        else:
+        max_line_number = file["lines"][-1]["line_number"] if file["lines"] else 1
+        try:
             with open(fname, "rb") as fh_in2:
                 source_lines = fh_in2.read().splitlines()
             lines = len(source_lines)
-            max_line_from_cdata = (
-                file["lines"][-1]["line_number"] if file["lines"] else 1
-            )
-            if lines < max_line_from_cdata:
+            if lines < max_line_number:
                 LOGGER.warning(
-                    f"File {fname} has {lines} line(s) but coverage data has {max_line_from_cdata} line(s)."
+                    f"File {max_line_number} has {lines} line(s) but coverage data has {max_line_number} line(s)."
                 )
-                # Python ranges are exclusive. We want to iterate over all lines, including
-                # that last line. Thus, we have to add a +1 to include that line.
-                for _ in range(lines, max_line_from_cdata):
-                    source_lines.append(b"/*EOF*/")
+                source_lines += [b"/*EOF*/"] * (max_line_number - lines)
+        except OSError as e:
+            if file["file"].endswith("<stdin>"):
+                message = f"Got sourcefile {file['file']}, using empty lines."
```

Comment:
`Got unreadable source file '{file['file']}', replacing with empty lines.`

Reply:
No replacement in f-string?

Reply:
yes, you can f-string it, I was just suggesting a clearer message

Reply:
Ah, there is a Icon for suggestions if you add a comment. In that case I get a diff and only need to accept.

Reply:
Ah, there is a Icon for suggestions if you add a comment. In that case I get a diff and only need to accept.